### PR TITLE
VimEditingEngine: Add 3 Vim bindings to insert mode

### PR DESCRIPTION
**Pull request by arieldon**
Add CTRL-W, CTRL-H, and CTRL-U in insert mode to Vim emulation. These key combinations revolve around deletion in insert mode.

- CTRL-W deletes the previous word
- CTRL-H deletes the previous character
- CTRL-U deletes the characters from the beginning of the line to the cursor

### CODE REVIEW

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 1]
```diff
@@ -884,6 +884,19 @@ void TextEditor::keydown_event(KeyEvent& event)
...
 
+void TextEditor::delete_previous_word()
+{
+    TextPosition word_break = document().first_word_break_before(m_cursor, true);
+    TextRange to_erase(word_break, m_cursor);
+
+    if (document().text_in_range(to_erase).trim_whitespace().length() == 0) {
+        word_break = document().first_word_break_before(word_break, true);
+        to_erase.set_start(word_break);
+    }
```
Comment (sin-ack) [inline Userland/Libraries/LibGUI/TextEditor.cpp:895]: Shouldn't `first_word_break_before` skip the whitespace correctly the first time?

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 2]
```diff
@@ -906,6 +919,30 @@ void TextEditor::delete_current_line()
...
+    execute<RemoveTextCommand>(document().text_in_range(to_erase), to_erase);
+}
+
+void TextEditor::delete_from_line_start_to_cursor()
+{
+    TextPosition start(m_cursor.line(), current_line().first_non_whitespace_column());
+    TextRange to_erase(start, m_cursor);
+    execute<RemoveTextCommand>(document().text_in_range(to_erase), to_erase);
+}
+
```
Comment (sin-ack) [inline Userland/Libraries/LibGUI/TextEditor.cpp:945]: Things like this are already implemented as Vim motions, which makes me think whether we can expose those for usage without requiring the user to enter them. Not a requirement for this PR, just a thought.

File: Userland/Libraries/LibGUI/TextEditor.cpp (hunk 1 above)
Comment (arieldon) [reply Userland/Libraries/LibGUI/TextEditor.cpp:895]: As in, I should move this logic into `first_word_break_before` itself?

Comment (sin-ack) [reply Userland/Libraries/LibGUI/TextEditor.cpp:895]: Perhaps a `first_word_before` for `TextDocument` could be nicer. I conflated `word_break` with `word`, my bad.

**Review by sin-ack — APPROVED** [review]

**Comment by gunnarbeutner** [comment]
The commit messages should probably start with "LibGUI:". Looks good to me otherwise.

**Pull request merged**